simplify tox.ini, .travis.yml, .coveragerc

zancas opened the pull request:
Issues:
Fixes #631

Problem: Each configuration file used by the test infrastructure
should have a discrete, non-overlapping function.

Analysis: This commit partitions responsibilities among several
config files:
* .travis.yml -- CI interface, specifies which test types to run
* tox.ini -- specifies environment and commands for a given test type
* requirements.TYPE.txt -- manifests for different test environments
requirements.TYPE.txt are used by tox.
* systest/Makefile -- buildbot interface, used by bb to run tests
by type, uses tox to configure test environments, outputs formatted
results in standard locations for consumption by trtl

Tests: The full nightly regression suites were run in the bbot
sandbox to verify correctness.   The github based CI system also
ran successfully before this commit was accepted.


NOTE:  I removed `.coveragerc`!!!  Does this break some tool I'm not aware of?

mattgreene commented:
@zancas : You need to ask the team for context prior to opening a PR with the following:
NOTE: I removed .coveragerc!!! Does this break some tool I'm not aware of?

This is not acceptable since we have team goals to collect code coverage.  @pjbreaux uses this during the gap analysis.  I set this up months ago and there are multiple .coveragerc files to collect data the various ways we run nightly tests.

zancas commented:
I've removed the removal, and addressed comments here:

https://github.com/zancas/f5-openstack-agent/pull/1

I like the combination of updatable diffs with a reviewer comment thread.  I believe that making the PR be within my own fork is a good "Have Our Cake And Eat it Too Solution" that is both powerful and safe.

I've sent invitations to that project to facilitate the review discussion.
